source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -402,25 +430,245 @@ struct functor_add_vj {
+                  continue;
+              }
+              stop_time.journey_pattern_point = *it;
+              stop_times.push_back(stop_time);
+          }
+          // calling the move constructor
+          vj_ptr->stop_time_list = std::move(stop_times);
+
+          auto& vj_list = get_vj_list_of_jp_helper<VJ_T>(old_jp);
+
+          auto it_old_vj = std::find_if(std::begin(vj_list), std::end(vj_list),
+                  [&vj_ptr](const std::unique_ptr<VJ_T>& old_vj ){
+                  return old_vj->uri == vj_ptr->uri;
+          });
+
+          assert(it_old_vj!= std::end(vj_list));
+
+          // Detach the vj from its jp and give it's ownership to a temporary unique_ptr
+          std::unique_ptr<VJ_T> old_vj_unique_ptr{it_old_vj->release()};
+          vj_list.erase(it_old_vj);
+
+          // Assign the new_jp to the vj
+          vj_ptr->journey_pattern = new_jp;
+
+          // We should also remove the old jp form impact's impacted_journey_patterns
+          boost::remove_erase_if(impact->impacted_journey_patterns, [&](const nt::JourneyPattern* jp){
+              return jp->uri == old_jp->uri;
+          });
+
+          // Now we can attach the vj to the  new jp
+          get_vj_list_of_jp_helper<VJ_T>(new_jp).push_back(std::move(old_vj_unique_ptr));
+
+      }
+  }
+
+  // update modified vj with its new jp to attach
+  void complete() {
+      complete_impl(vj_discrete_to_be_updated);
+      complete_impl(vj_frequency_to_be_updated);
+  };
+
+  bool is_impacted(const nt::VehicleJourney& vj) const{
+      for (auto period : impact->application_periods) {
+          bg::day_iterator titr(period.begin().date() - bg::days(1));
+          for(;titr<=period.end().date(); ++titr) {
+              if (!meta.production_date.contains(*titr)) {
+                  continue;
+              }
+              auto day = (*titr - meta.production_date.begin()).days();
+
+              if (!vj.is_adapted) {
+                  if (vj.validity_pattern->check(day)) {
+                      return true;
+                  }
+              }else{
+                  // TODO: We should handle the intersected disruption periods
+                  if (vj.adapted_validity_pattern->check(day)) {
+                      return true;
+                  }
+              }
+
+          }
+      }
+      return false;
+  }
+
+  nt::JourneyPattern* get_or_create_journey_pattern(const nt::JourneyPattern& adapted_jp) {
+      std::vector<nt::StopPoint*> stop_points_for_key{};
+      for (const auto* jpp: adapted_jp.journey_pattern_point_list) {
+          if (! contains(impacted_stop_points, jpp->stop_point)) {
+              stop_points_for_key.push_back(jpp->stop_point);
+          }
+      }
+      auto jp_key = nt::JourneyPatternKey{adapted_jp, std::move(stop_points_for_key)};
+      return pt_data.get_or_create_journey_pattern(jp_key);
+  }
+
+  template<typename T>
+  bool is_updated_with_impacted_stop_points(const T& vj){
```

Comment:
Add hour check for is_impacted here maybe? (linked to remark above on correct day)